akka-actor/src/main/scala/akka/io/DirectByteBufferPool.scala

```diff
@@ -36,32 +36,32 @@ private[akka] class DirectByteBufferPool(defaultBufferSize: Int, maxPoolEntries:
   private def allocate(size: Int): ByteBuffer =
     ByteBuffer.allocateDirect(size)
 
-  @tailrec
-  private final def takeBufferFromPool(): ByteBuffer =
-    if (locked.compareAndSet(false, true)) {
-      val buffer =
-        try if (buffersInPool > 0) {
+  private final def takeBufferFromPool(): ByteBuffer = {
+    var buffer: ByteBuffer = null
+
+    if (lock.tryLock(1, TimeUnit.MILLISECONDS))
```

Comment:
There’s no point in blocking this thread longer than 1ms for an optimization that we can do well without. If we can’t access the pool right now we’ll have to allocate a new buffer.